packages/flutter/test/painting/box_painter_test.dart

```diff
@@ -107,7 +109,160 @@ void main() {
     expect(shadowList, equals(<BoxShadow>[shadow4, shadow1.scale(0.5)]));
   });
 
+  test('BoxShadow BlurStyle test', () {
+    const BoxShadow shadow1 = BoxShadow(blurRadius: 4.0);
+    const BoxShadow shadow2 = BoxShadow(blurRadius: 4.0, blurStyle: BlurStyle.outer);
+    final BoxShadow shadow3 = BoxShadow.lerp(shadow1, null, 0.25)!;
+    final BoxShadow shadow4 = BoxShadow.lerp(null, shadow1, 0.25)!;
+    final BoxShadow shadow5 = BoxShadow.lerp(shadow1, shadow2, 0.25)!;
+    final BoxShadow shadow6 = BoxShadow.lerp(const BoxShadow(blurStyle: BlurStyle.solid), shadow2, 0.25)!;
+
+    expect(shadow1.blurStyle, equals(BlurStyle.normal));
+    expect(shadow2.blurStyle, equals(BlurStyle.outer));
+    expect(shadow3.blurStyle, equals(BlurStyle.normal));
+    expect(shadow4.blurStyle, equals(BlurStyle.normal));
+    expect(shadow5.blurStyle, equals(BlurStyle.outer));
+    expect(shadow6.blurStyle, equals(BlurStyle.solid));
+
+    List<BoxShadow> shadowList = BoxShadow.lerpList(<BoxShadow>[shadow2, shadow1], <BoxShadow>[shadow3], 0.5)!;
+    expect(shadowList[0].blurStyle, equals(BlurStyle.outer));
+    expect(shadowList[1].blurStyle, equals(BlurStyle.normal));
+
+    shadowList = BoxShadow.lerpList(<BoxShadow>[shadow6], <BoxShadow>[shadow3, shadow1], 0.5)!;
+    expect(shadowList[0].blurStyle, equals(BlurStyle.solid));
+    expect(shadowList[1].blurStyle, equals(BlurStyle.normal));
+
+    shadowList = BoxShadow.lerpList(<BoxShadow>[shadow3], <BoxShadow>[shadow6, shadow1], 0.5)!;
+    expect(shadowList[0].blurStyle, equals(BlurStyle.solid));
+    expect(shadowList[1].blurStyle, equals(BlurStyle.normal));
+
+    shadowList = BoxShadow.lerpList(<BoxShadow>[shadow3], <BoxShadow>[shadow2, shadow1], 0.5)!;
+    expect(shadowList[0].blurStyle, equals(BlurStyle.outer));
+    expect(shadowList[1].blurStyle, equals(BlurStyle.normal));
+  });
+
   test('BoxShadow toString test', () {
-    expect(const BoxShadow(blurRadius: 4.0).toString(), equals('BoxShadow(Color(0xff000000), Offset(0.0, 0.0), 4.0, 0.0)'));
+    expect(const BoxShadow(blurRadius: 4.0).toString(), equals('BoxShadow(Color(0xff000000), Offset(0.0, 0.0), 4.0, 0.0), BlurStyle.normal'));
+    expect(const BoxShadow(blurRadius: 4.0, blurStyle: BlurStyle.solid).toString(), equals('BoxShadow(Color(0xff000000), Offset(0.0, 0.0), 4.0, 0.0), BlurStyle.solid'));
+  });
+
+  testWidgets('BoxShadow BoxStyle.solid', (WidgetTester tester) async {
+    final Key key = UniqueKey();
+    await tester.pumpWidget(
+      Center(
+        child: RepaintBoundary(
+          key: key,
+          child: Container(
+            color: Colors.white,
+            width: 50,
+            height: 50,
+            child: Center(
+              child: Container(
+                decoration: const BoxDecoration(
+                  boxShadow: <BoxShadow>[BoxShadow(blurRadius: 3.0, blurStyle: BlurStyle.solid)],
```

Comment:
Let's also use a wider radius! Simply easier for us to visually inspect the results.